composer/cli/launcher.py

```diff
@@ -107,30 +120,87 @@ def parse_args():
     parser = get_parser()
 
     args = parser.parse_args()
-    if args.world_size == -1:
+
+    env_var_mapping = {
```

Comment:
Thoughts about setting the defaults in the argparse above? E.g. something like `default=int(os.environ.get('BASE_RANK', '0')`? For world size, could default it to None, since the default is thr value of nproc. Just not a big fan of `getattr` and `setattr` inside the for loop.

Reply:
Not a huge fan of putting complicated defaults in, since these will be displayed verbatim in our docs. I'm just going to rewrite this to not use a for loop, and instead be a bit verbose.